Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -56,6 +57,11 @@ public object Deserialize(IIonReader reader)
 
                     field.SetValue(targetObject, deserialized);
                 }
+                else if ((method = FindSetterMethod(reader.CurrentFieldName)) != null)
```

Comment:
As this is method getting bigger, can you please add comments for each block here to indicate at a high level what they are targeting?

Reply:
Comments added.